backend/src/services/user/user.service.ts

```diff
@@ -41,6 +45,52 @@ class User {
       email: user.email,
     };
   }
+
+  async getByEmail(email: string): Promise<UserSignUpResponseDto | null> {
+    const user = await this.#userRepository.getByEmail(email);
+
+    if (!user) {
+      return null;
+    }
+
+    return {
+      id: user.id,
+      email: user.email,
+    };
+  }
+
+  async getPasswordHashByEmail(
+    email: string,
+  ): Promise<UserPasswordHashDto | null> {
+    const user = await this.#userRepository.getByEmail(email);
+
+    if (!user) {
+      return null;
+    }
+
+    return {
+      id: user.id,
+      email: user.email,
+      passwordHash: user.passwordHash,
+    };
+  }
```

Comment:
```suggestion
```

Reply:
@what1s1ove do you mean I have to get password hash in auth service via user.getByEmail ?